includes/discovery/sensors/fanspeed/benu.inc.php

```diff
@@ -0,0 +1,14 @@
+<?php
+
+// Adapated from Bluecoat sgos discovery
+
+echo 'Benu Sensors ';
+$sensor_index = 0;
+for ($index = 4; $index <= 9; $index++) { //Benu Fans are index 4 thru 9
+    $descr_oid   = "benuSensorName.1.$index";
+    $fan_oid     = ".1.3.6.1.4.1.39406.1.1.1.4.1.1.5.1.$index";
+    $descr       = snmp_get($device, $descr_oid, '-Oqv', 'BENU-CHASSIS-MIB');
+    $current     = snmp_get($device, $fan_oid, '-Oqv', 'BENU-CHASSIS-MIB');
```

Comment:
The issue that I'm facing currently is when I use the name 'benuSensorValue' rather than the number the poller module can't find it. 
SNMP[/usr/bin/snmpget -v2c -c COMMUNITY -OUQnt -M /opt/librenms/mibs:/opt/librenms/mibs/benuos udp:HOSTNAME:161 benuSensorValue.1.10 benuSensorValue.1.11]
benuSensorValue.1.10: Unknown Object Identifier (Sub-id not found: (top) -> benuSensorValue)
benuSensorValue.1.11: Unknown Object Identifier (Sub-id not found: (top) -> benuSensorValue)


If I get what your saying in the code I can have the name but in the when I register it needs the numerical?
I would have two declarations for the OID
  $fan_oid_name = "benuSensorValue.1.$index"
  $fan_oid_numb = ".1.xxxxxx.1.$index"
Then in the discover_sensor function I reference $fan_oid_num?


It's been a long time since I did C so forgive the stupid question.
Are the input's and outputs of snmp_get_multi an array? Something like this?

$sensor_index = "1,2";
$descr_oids   = "benuSensorName.1.10,benuSensorName.1.11";
$volt_oids    = "benuSensorValue.1.10,benuSensorValue.1.11";

$descr   = function snmp_get_multi($device, $descr_oids, $options = '-OQUs', 'BENU-CHASSIS-MIB');
$current = function snmp_get_multi($device, $volt_oids, $options = '-OQUs', 'BENU-CHASSIS-MIB');

Can I feed the array into the discover function?

discover_sensor($valid['sensor'], 'voltage', $device, $volt_oids, $sensor_index, 'benuos', $descr, '1', '1', null, null, null, null, $current);